### PR TITLE
eve/nfs: switch output to jsonbuilder

### PR DESCRIPTION
**Pull request by victorjulien**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3711

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
Passed

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by victorjulien** [comment]
CI failure is not caused by this PR
```
output-json-alert.c:476:40: error: missing field 'state_index' initializer
2062      [-Werror,-Wmissing-field-initializers]
2063            JsonBuilderMark mark = { 0 };
2064
```

**Comment by jasonish** [comment]
> 2063            JsonBuilderMark mark = { 0 };

Isn't this valid C99 to initialize a struct to 0's?

**Comment by victorjulien** [comment]
> > 2063            JsonBuilderMark mark = { 0 };
> 
> Isn't this valid C99 to initialize a struct to 0's?

I don't know. Googling doesn't give a clear answer, so lets assume not.

File: rust/src/nfs/log.rs [hunk 1]
```diff
@@ -35,43 +35,43 @@ pub extern "C" fn rs_nfs_tx_logging_is_filtered(state: &mut NFSState,
     return 0;
 }
 
-fn nfs_rename_object(tx: &NFSTransaction) -> Json
+fn nfs_rename_object(tx: &NFSTransaction, js: &mut JsonBuilder)
+    -> Result<bool, JsonError>
```
Comment (jasonish) [inline rust/src/nfs/log.rs:39]: As this function either errors out or returns true, and the Ok(val) is never checked, its probably better as `Result<(), JsonError>`.

If you pulled this from any of my examples, oops, I had meant to get rid of them.

Comment (victorjulien) [reply rust/src/nfs/log.rs:39]: My bad, I pulled it from a place in dns where it does serve a purpose. Fixed in #5023

**Comment by victorjulien** [comment]
> Overall looks good.
> 
> What did you think of doing the conversion? Any oddities?

No very easy.

**Comment by victorjulien** [comment]
Replaced by #5023

**Pull request closed without merging**